strategy/src/lib.rs

```diff
@@ -0,0 +1,407 @@
+//!
+//!     fn poll_with_strategy<S: Strategy>(
+//!         mut self: Pin<&mut Self>,
+//!         strategy: &mut S,
+//!         context: &mut S::Context,
+//!     ) -> Poll<Self::Output> {
+//!         match strategy.poll(self.listener.take().unwrap(), context) {
+//!             Ok(()) => Poll::Ready(()),
+//!             Err(listener) => {
+//!                 self.listener = Some(listener);
+//!                 Poll::Pending
+//!            }
+//!        }
+//!     }
+//! }
+//!
+//! // Use the future in a blocking context.
+//! let future = wait_three_seconds();
+//! future.wait();
+//!
+//! // Use the future in a non-blocking context.
+//! futures_lite::future::block_on(async {
+//!     let future = FutureWrapper::new(wait_three_seconds());
+//!     future.await;
+//! });
+//! ```
+
+#![cfg_attr(not(feature = "std"), no_std)]
+#![cfg_attr(docsrs, feature(doc_cfg))]
+#![forbid(unsafe_code, future_incompatible, missing_docs)]
+
+use core::future::Future;
+use core::marker::PhantomData;
+use core::pin::Pin;
+use core::task::{Context, Poll};
+
+use event_listener::EventListener;
+use pin_utils::pin_mut;
+
+/// A wrapper around an [`EventListenerFuture`] that can be easily exported for use.
+///
+/// This type implements [`Future`], has a `_new()` constructor, and a `wait()` method
+/// that uses the [`Blocking`] strategy to poll the future until it is ready.
+///
+/// # Examples
+///
+/// ```
+/// mod my_future {
+///     use event_listener_strategy::{easy_wrapper, EventListenerFuture, Strategy};
+///     use std::pin::Pin;
+///     use std::task::Poll;
+///
+///     struct MyFuture;
+///
+///     impl EventListenerFuture for MyFuture {
+///         type Output = ();
+///
+///         fn poll_with_strategy<S: Strategy>(
+///             self: Pin<&mut Self>,
+///             strategy: &mut S,
+///             context: &mut S::Context,
+///         ) -> Poll<Self::Output> {
+///             /* ... */
+/// #           Poll::Ready(())
+///         }
+///     }
+///
+///     easy_wrapper! {
+///         /// A future that does something.
+///         pub struct MyFutureWrapper(MyFuture => ());
+///         /// Wait for it.
+///         pub wait();
+///     }
+///
+///     impl MyFutureWrapper {
+///         /// Create a new instance of the future.
+///         pub fn new() -> Self {
+///             Self::_new(MyFuture)
+///         }
+///     }
+/// }
+///
+/// use my_future::MyFutureWrapper;
+///
+/// // Use the future in a blocking context.
+/// let future = MyFutureWrapper::new();
+/// future.wait();
+///
+/// // Use the future in a non-blocking context.
+/// futures_lite::future::block_on(async {
+///     let future = MyFutureWrapper::new();
+///     future.await;
+/// });
+/// ```
+#[macro_export]
+macro_rules! easy_wrapper {
+    (
+        $(#[$meta:meta])*
+        $vis:vis struct $name:ident ($inner:ty => $output:ty);
+        $(#[$wait_meta:meta])*
+        $wait_vis: vis wait();
+    ) => {
+        ::pin_project_lite::pin_project! {
```

Comment:
this requires a dependency on `pin-project-lite` at the usage site, is that intended? (alternative would be re-exporting `pin_project_lite` in this crate (perhaps with `#[doc(hidden)]`), and prefixing this with `$crate`